client/test/error-telemetry.test.js

```diff
@@ -0,0 +1,45 @@
+jest.mock('fs');
+
+const assert         = require('assert');
+const fs             = require('fs');
+const ErrorTelemetry = require('../src/lib/error-telemetry');
+const mkdirp         = require('mkdirp');
+
+describe('Error Telemetry Logging', () => {
+  beforeEach(() => {
+    /* Make sure memfs is flushed every time */
+    fs.volume.reset();
+  });
+
+  describe('setup() call', () => {
+
+    // FIXME: only hackish, the end goal is definitely not to forward to another file
+    it('writing to essentials logging file should forward to another', done => {
+
+      // Write before setup to make sure the file is already present
+      const logsDir = '/evergreen/jenkins/var/logs/';
+      const logFile = logsDir + 'essentials.log.0';
+      mkdirp.sync(logsDir);
+      fs.writeFileSync(logFile, '{"timestamp":1523451065975,"level":"SEVERE","message":"WAT"}\n');
+
+      assert(fs.existsSync(logFile));
+
+      mkdirp.sync('/tmp');
+      const response = new ErrorTelemetry().setup();
+      assert(!(response instanceof Promise));
+
+      assert(!fs.existsSync('/tmp/test'));
+
+      setTimeout( () => {
```

Comment:
I'm not sure how jest works with `setTimeout`, you may want to consider using [Timer mocks](https://facebook.github.io/jest/docs/en/timer-mocks.html)

Although, I'm not sure if that's necessary for this initial kind of hacky test case

Reply:
Yeah, and I've tried quite a bit. Possibly I missed something, but couldn't make it work for now. We ended up removing them when hacking with @Wadeck mentoring.

I think I'll try to reintroduce them when I'm more used to the wholee thing.